appsec-kit-v7/src/main/java/com/vaadin/appsec/v7/service/NotificationInitializer.java

```diff
@@ -56,32 +63,36 @@ static void serviceInit(VaadinService vaadinService) {
         if (isDebugMode(vaadinService)) {
             vaadinService.addSessionInitListener(e -> {
                 e.getSession().addUIProvider(new AppSecUIProvider());
-                createNotificationThread(e.getSession()).start();
+                startNotificationTask(e.getSession());
             });
             LOGGER.info("NotificationInitListener initialized.");
         }
     }
 
-    private static Thread createNotificationThread(
-            final VaadinSession session) {
-        return new Thread(() -> {
-            try {
-                LOGGER.info(
-                        "NotificationInitListener notification thread initialized.");
-
-                Thread.sleep(NOTIFICATION_CHECK_INTERVAL);
-
-                while (isSessionOpen(session)) {
-                    session.access(() -> {
-                        session.getUIs()
-                                .forEach(ui -> notifyUiIfNeeded(session, ui));
-                    });
-                    Thread.sleep(NOTIFICATION_CHECK_INTERVAL);
-                }
-            } catch (InterruptedException e) {
-                // NOP
-            }
-        });
+    private static void startNotificationTask(final VaadinSession session) {
+        final String sessionId = session.getSession().getId();
+        final ScheduledFuture<?> scheduledNotificationCheck = AppSecService
+                .getInstance().getConfiguration().getTaskExecutor()
+                .scheduleAtFixedRate(() -> {
+                    LOGGER.info(
+                            "NotificationInitListener notification thread initialized.");
```

Comment:
Should we log this message each time the task runs?

Reply:
Definitely not. Good catch!

Reply:
Fixed